### PR TITLE
Remove handling of dependencies via packrat in R APIs

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?
 
In R, MLflow currently uses packrat to package & restore R dependencies to/from `r-dependencies.txt` files for projects & models, analogous to `conda.yaml`. Post-1.0 we'd like to replace packrat with `renv` which will contain more advanced support for e.g. packaging conda dependencies. This PR therefore moves removes packrat support for MLflow 1.0 by removing the `mlflow_snapshot`, `mlflow_restore_snapshot` APIs and elminating(where present) the `r_dependencies` argument used to specify the path to a packrat r-dependencies.txt file.

This is a breaking API change.

## How is this patch tested?
 
Existing R unit tests
 
### Is this a user-facing change? 

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
* The `mlflow_snapshot` and `mlflow_restore_snapshot` APIs have been removed
* New R APIs will be introduced after MLflow 1.0 for managing code and model dependencies
* The `r_dependencies` argument used to specify the path to a packrat r-dependencies.txt file has been removed from all APIs.

### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [ ] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [x] Scoring 
- [x] Serving
- [x] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:
 
- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/R/mlflow/R/model.R [hunk 1]
```diff
@@ -3,23 +3,20 @@
...
 #'   will be saved.
-#' @param r_dependencies Optional vector of paths to dependency files
-#'   to include in the model, as in \code{r-dependencies.txt}
-#'   or \code{conda.yaml}.
 #' @param conda_env Path to Conda dependencies file.
 #'
 #' @importFrom yaml write_yaml
 #' @export
-mlflow_save_model <- function(x, path = "model", r_dependencies=NULL, conda_env=NULL) {
+mlflow_save_model <- function(model, path = "model", conda_env=NULL) {
```
Comment (kevinykuo) [inline mlflow/R/mlflow/R/model.R]: We can probably remove the `conda_env` here and also from the signature of `mlflow_save_flavor()` since not all flavors care about conda. We can use `...` and forward it to the `mlflow_save_flavor()` call.

File: mlflow/R/mlflow/R/model.R [hunk 2]
```diff
@@ -77,14 +78,15 @@ mlflow_predict_model <- function(model, data) {
...
 #' @template roxlate-model-uri
+#' @template roxlate-client
 #' @param flavor Optional flavor specification. Can be used to load a particular flavor in case
 #'        there are multiple flavors available.
 #' @export
 mlflow_load_model <- function(model_uri, flavor = NULL, client = mlflow_client()) {
   model_path <- mlflow_download_artifacts_from_uri(model_uri, client = client)
   supported_flavors <- supported_model_flavors()
   spec <- yaml::read_yaml(fs::path(model_path, "MLmodel"))
-  available_flavors <- intersect(names(spec$flavors), supported_flavors)
```
Comment (tomasatdatabricks) [inline mlflow/R/mlflow/R/model.R]: Why did you remove intersection with supported flavors? How do we know which flavors are available in R?

**Review by tomasatdatabricks — COMMENTED** [review]
Thanks for the PR!  2 questions:

1. why did you remove intersection with available flavors when loading the model?
2. what about r-dependencies in crate? those are currently not used, should we remove them?

File: mlflow/R/mlflow/R/model.R (hunk 2 above)
Comment (smurching) [reply mlflow/R/mlflow/R/model.R]: Oops good question, accidentally included that change in this PR - it's meant to be included in https://github.com/mlflow/mlflow/pull/1269, will discuss it there & remove the change from this PR

**Comment by smurching** [comment]
Good point regarding 2), yeah happy to remove `r_dependencies` from crate and keras models as well in this PR (basically I'll just revert https://github.com/mlflow/mlflow/pull/1263/commits/fe5bb712968b58e3345b3cf407e050d2f382c127,I was mistakenly worried about conflicts with #1257)

**Review by tomasatdatabricks — APPROVED** [review]
lgtm
